hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Hangman.Game do
+
+    defstruct(
+        turns_left: 7, 
+        game_state: :initializing,
+        word: [],
+        letters: [],
+        used: [],
+        last_guess: ""
+    )
+
+#####################################################################
+
+    def new_game() do
+        new_word = Dictionary.random_word()
+        |> String.trim("\r")
+        |> String.graphemes()
+
+        %Hangman.Game{
+            word: new_word, 
+            letters: List.duplicate("_",length(new_word))
+        }
+    end
+
+#####################################################################
+
+    def tally(game), do: Map.delete(game,:word)
+
+#####################################################################
+
```

Comment:
This note covers the whole function.

First, I wondered why all the single character variable names. In general, I really hate these for anything apart from block-locals.

Then I saw you had an 8 parameter function call, and the light went on. You needed short names so it would fit.

But, I think you fixed the wrong problem. The issue wasn't verbosity, but the `guess_attempt/8` function.

Rather than working out all the possible interesting values up-front, why not implement more like a decision tree. Have the top level function make a simply determination (for example, has the letter already been used). Then use this when calling the next-level down function. This in turn makes one more determination, and delegates down again.

You'd end up with 3 or 4 functions, rather than this single function, but most of your `guess_attempt` heads would disappear.

## -8